### PR TITLE
[ADD]: Client registration APIs

### PR DESCRIPTION
**Pull request by nishihere19**
Added:-
Client Registration API
Fetch list of clients for a particular resource owner
Fetch all the details of a client using the client id for a user.

### CODE REVIEW

File: src/routes/client.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
+import express, { Router, Request } from 'express';
+import { isAuthenticated } from '../controllers/authContoller';
+import {
+  getClientDetailsByUser,
+  getClientList,
+  registerClient,
+  validateClientRegisterFields
+} from '../controllers/clientController';
+
```
Comment (siva2204) [inline src/routes/client.ts:9]: have isAuthenticated as global middleware for this module ?

File: src/controllers/clientController.ts [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+): Promise<unknown> => {
+  const errors = validationResult(req);
+  if (!errors.isEmpty()) {
+    return res.status(400).json({ errors: errors.array() });
+  }
+
+  const { name, homepageUrl, callbackUrl, description } = req.body;
+  try {
+    // generate client credentials
+    const clientId = await generateClientId();
```
Comment (siva2204) [inline src/controllers/clientController.ts:49]: ~await~

File: src/controllers/clientController.ts [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+    const clientSecret = await generateClientSecret();
```
Comment (siva2204) [inline src/controllers/clientController.ts:50]: ~await~

File: src/controllers/clientController.ts (hunk 2 above)
Comment (nishihere19) [reply src/controllers/clientController.ts]: Applied the suggested changes

**Review by siva2204 — APPROVED** [review]

**Pull request merged**